_posts/2025-03-26-phased-strimzi-upgrade-example.md

```diff
@@ -0,0 +1,700 @@
+- 045-Crd-kafkamirrormaker.yaml
+- 046-Crd-kafkabridge.yaml
+- 047-Crd-kafkaconnector.yaml
+- 048-Crd-kafkamirrormaker2.yaml
+- 049-Crd-kafkarebalance.yaml
+- 04A-Crd-kafkanodepool.yaml
+```
+
+Contents of `install-files/crds/kustomization.yaml`
+```yaml
+resources:
+- cluster-operator-0.43.0
+- crds
+```
+
+Using the setup above we can deploy the Strimzi 0.43 operator using the single command:
+
+```shell
+kubectl -n <strimzi namespace> apply -k install-files
+```
+
+This will tell `kubectl` to look in the `install-files` directory for a `kustomization.yaml` file and import whatever resources and patches it finds there. 
+This in-turn cascades through each of the folders defined in that top level kustomization and runs those imports and patches and so on.
+
+If you want to see the output sent to Kubernetes you can preview it by running:
+
+```shell
+kubectl kustomize install-files
+```
+
+This will render the kustomize altered install resources to the console.
+
+##### Renaming Resources
+
+Now we have a deployment for a single Strimzi version we need to add another. 
+As discussed above however, we need to rename the RBAC and configuration resources so each version has its own set. 
+Kustomize makes this easy using the `nameSuffix` patch. 
+This adds the supplied string to the name of every resource in the `resources` list defined in that `kustomization.yaml` file. 
+It also updates any references to those renamed resources.
+For example this will change the name of the Strimzi configuration `ConfigMap` resource and also the reference to it in the Strimzi Cluster Operator `Deployment` resource.
+
+To make this change we simply add one line to the `kustomization.yaml` in each Strimzi version directory:
+
+`install-files/cluster-operator-0.43.0/kustomization.yaml`
+```yaml
+resources:
+- 010-ServiceAccount-strimzi-cluster-operator.yaml
+...
+- 060-Deployment-strimzi-cluster-operator.yaml
+
+nameSuffix: '-0-43-0'
+```
+
+That's it. 
+You can see how it changes the names by building the kustomization with:
+```shell
+kubectl kustomize install-files
+```
+
+We can now add a newer Strimzi version by downloading the install files for that version, moving the CRDs to the separate `crds` folder (replacing the older ones) and then adding a `kustomization.yaml` to that version's folder with a different `nameSuffix`:
+
+`install-files/cluster-operator-0.44.0/kustomization.yaml`
+```yaml
+resources:
+- 010-ServiceAccount-strimzi-cluster-operator.yaml
+...
+- 060-Deployment-strimzi-cluster-operator.yaml
+
+nameSuffix: '-0-44-0'
+```
+
+##### Patching Environment Variables
+
+Sadly, the `nameSuffix` is not sufficient on its own. 
+There are several other elements of the deployment files that need to be customised in order to allow the Strimzi versions to exist concurrently.
+
+The first of these are the environment variables in the Strimzi operator `Deployment` resource, several of which need to be individualised to each Strimzi version. 
```

Comment:
Not sure about this _individualised_ in the doc
I might go for something like tailoring or customizing

Reply:
I agree, _tailored_ feels cleaner.